internal/gps/source_manager.go

```diff
@@ -519,24 +518,13 @@ func (sm *SourceMgr) InferConstraint(s string, pi ProjectIdentifier) (Constraint
 		return Any(), nil
 	}
 
-	slen := len(s)
-	if slen == 40 {
-		if _, err := hex.DecodeString(s); err == nil {
-			// Whether or not it's intended to be a SHA1 digest, this is a
-			// valid byte sequence for that, so go with Revision. This
-			// covers git and hg
-			return Revision(s), nil
-		}
-	}
-
-	// Next, try for bzr, which has a three-component GUID separated by
-	// dashes. There should be two, but the email part could contain
-	// internal dashes
+	// Bazaar has a three-component GUID separated by dashes. There should be two,
```

Comment:
What happens when you remove the specific handling for bzr? I was hoping (but don't know for sure!) that `repo.CommitInfo` would work for all vcs types, including bzr.

Reply:
Yeah, I don't think our CI build runs against any real bzr repos?

Reply:
the gps tests include some [real](https://github.com/golang/dep/blob/f1346974068313371d889e120ec9e01f5ba43642/internal/gps/vcs_repo_test.go#L352) [tests](https://github.com/golang/dep/blob/f1346974068313371d889e120ec9e01f5ba43642/internal/gps/vcs_source_test.go#L285) against bzr.